### ISSUE TITLE
Realtime subscriptions are delivered out of order

### ISSUE THREAD

**Issue by xrd** [pocketbase/pocketbase#590]
Perhaps I don't understand exactly how realtime subscriptions work, but this seems odd to me. When I use code like this:

```
client.realtime.subscribe("words", async function(e) { console.log('Updating/creating', e); } );
```

I see in my console that the `e.action` field shows the updated before the created event.

I'm using a customized framework app, with `OnModelAfterCreate()` that annotates the record, and it actually kind of amazes me that my remote API call completes, updates the record, and then delivers an updated event to the realtime handler before the create event is delivered! But, perhaps because I'm doing this extra work, the events are delivered out of order?

I can always write my own client code to determine what's the best course of action, ignoring if the item with the same ID is already in my set if the action is `created`. But, it seems strange that this should happen and probably a bug, right?

![update-create-realtime](https://user-images.githubusercontent.com/17064/191399098-6a7ed623-3d12-4415-9986-8899a79b4713.png)

**Comment by ganigeorgiev** [comment]
@xrd Could you provide a stripped-down version of your attached hooks? By "_my remote API call completes, updates the record_", do you mean that you are updating the same record in the OnModelAfterCreate hook?

The realtime create/update events are currently attached to the `OnModelAfterCreate` and `OnModelAfterUpdate` hooks, so it is possible if you are doing update inside the create hook, that event to get triggered before the create one. You can disable the default Dao hooks by creating a new one, from the model event (I'll consider changing this in v0.8.0 this to  be the default):
```
app.OnModelAfterCreate().Add(func(e *core.ModelEvent) error {
    dao := daos.New(e.Dao.DB())
    
    // use the above dao to apply your model changes

    return nil
})
```

Alternatively you could also attach to the `OnModelBeforeCreate` hook and modify the model fields with your external API call before the record is being persisted in the database.

**Comment by xrd** [comment]
I'll provide the code later but I'm not doing an update inside the call. I'm making an HTTP request to an outside server, then immediately finishing the hook. That external server does some processing and then uses the API to update the same record but in an unconnected stage of processing.

**Comment by ganigeorgiev** [comment]
@xrd Thanks for providing a code sample. I think I understand what actually may be the cause of the disordered events. I've submitted a fix in `master` that will ensure that the realtime api hooks are binded before the user defined ones, so in your case even if the `notify()` call takes some time it won't matter since it will be executed after the broadcasted event to the client.

Additionally, in order to reduce the checks and improve the latency, I've removed the `LogsDao()` events to avoid flooding the `OnModel*` hooks, since most users doesn't need them anyway. In the rare cases where you need to listen to the logs db changes, you can attach directly to the LogsDao(), by doing something like:

```go
app.LogsDao().BeforeCreateFunc = func(eventDao *daos.Dao, m models.Model) error {
	if requestModel, ok := m.(*models.Request); ok {
		redactLog(requestModel)
	}
	return nil
}
```

The fix is available currently only in `master` (https://github.com/pocketbase/pocketbase/commit/954067860c3e2cee2efe32a282e49e2219ea67bd) and it will be part of the next v0.7.6 bugfix release (sometime later this week).

If you don't want to wait, you can update your go mod dependency to get the master:
```
go get github.com/pocketbase/pocketbase@master
```

**Comment by ganigeorgiev** [comment]
@xrd Just to let you know that the above fix is now available in the latest [v0.7.6 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.7.6).
